### DISCUSSION TITLE
Repeat directive, this keyword is undefined.

### DISCUSSION THREAD

**Discussion by dbwodlf3** [lit/lit#2983] [General]
I am trying to repeat function with this keyword in lit element, but the "this" key word is undefined in handler function(method in class, typescript).
Of course it is not wrong, there are any tricky way to bypass this, but, just i would like to know...

**Comment by ynxtynxt** [comment]
Like rather than `myMethod( arg ){ this }` you can declare as `myMethod = ( arg ) => { this }` instead?

**Reply by dbwodlf3** [reply]
because of member variables. (like isShow or isEditable etc...)
```repeat(..., this.myHandler)``` instead of ```repeat( ... , (item)=>{...})```

**Reply by ynxtynxt** [reply]
Like in this [playground](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy5qcyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7cmVwZWF0fSBmcm9tICdsaXQvZGlyZWN0aXZlcy9yZXBlYXQuanMnO1xuXG5cbmV4cG9ydCBjbGFzcyBTaW1wbGVHcmVldGluZyBleHRlbmRzIExpdEVsZW1lbnQge1xuICBzdGF0aWMgc3R5bGVzID0gY3NzYHAgeyBjb2xvcjogYmx1ZSB9YDtcblxuICBzdGF0aWMgcHJvcGVydGllcyA9IHtcbiAgICBzZXA6IHt0eXBlOiBPYmplY3R9LFxuICB9O1xuXG4gIGNvbnN0cnVjdG9yKCkge1xuICAgIHN1cGVyKCk7XG4gICAgdGhpcy5zZXAgPSBodG1sYDxicj5gO1xuICB9XG5cbiAgbXlNZXRob2QoaXRlbSwgaW5kZXgpe1xuICAgIHJldHVybiBbaHRtbGAke2l0ZW19YCx0aGlzLnNlcF07XG4gIH1cblxuICBteUFycm93TWV0aG9kID0gKGl0ZW0sIGluZGV4KSA9PiB7XG4gICAgcmV0dXJuIFtodG1sYCR7aXRlbX1gLHRoaXMuc2VwXTtcbiAgfVxuXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGBcbiAgICAgICAgJHtyZXBlYXQoWycxJywnMicsJzMnXSwgdGhpcy5teU1ldGhvZC5iaW5kKHRoaXMpICl9XG4gICAgICAgICR7cmVwZWF0KFsnQScsJ0InLCdDJ10sIHRoaXMubXlBcnJvd01ldGhvZCApfWA7XG4gIH1cbn1cbmN1c3RvbUVsZW1lbnRzLmRlZmluZSgnc2ltcGxlLWdyZWV0aW5nJywgU2ltcGxlR3JlZXRpbmcpO1xuIn0seyJuYW1lIjoiaW5kZXguaHRtbCIsImNvbnRlbnQiOiI8IURPQ1RZUEUgaHRtbD5cbjxoZWFkPlxuICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCIuL3NpbXBsZS1ncmVldGluZy5qc1wiPjwvc2NyaXB0PlxuPC9oZWFkPlxuPGJvZHk-XG4gIDxzaW1wbGUtZ3JlZXRpbmcgbmFtZT1cIldvcmxkXCI-PC9zaW1wbGUtZ3JlZXRpbmc-XG48L2JvZHk-XG4ifSx7Im5hbWUiOiJwYWNrYWdlLmpzb24iLCJjb250ZW50Ijoie1xuICBcImRlcGVuZGVuY2llc1wiOiB7XG4gICAgXCJsaXRcIjogXCJeMi4wLjBcIixcbiAgICBcIkBsaXQvcmVhY3RpdmUtZWxlbWVudFwiOiBcIl4xLjAuMFwiLFxuICAgIFwibGl0LWVsZW1lbnRcIjogXCJeMy4wLjBcIixcbiAgICBcImxpdC1odG1sXCI6IFwiXjIuMC4wXCJcbiAgfVxufSIsImhpZGRlbiI6dHJ1ZX1d)?

**Reply by dbwodlf3** [reply]
yes. like that. I don't know why have to binding this. the method is declared in class namespace. (and arrow function is not method as i know..)
like [this](https://lit.dev/playground/#gist=5b7d87c128452e2ffc3e21801e696c6f). this is little bit weired.. i don't know why...

**Reply by ynxtynxt** [reply]
It isn't weird.. since you pass a reference to the function, you are not calling it in the same context, it's how js work not Lit's specific behavior. Only event listener expressions will automatically bind(this) eg. `@click=${ this.myAutoBoundMethod}`. And when you define a class field as an arrow function it's bound to the current context so you also use `this`.

Like you can `Promise.resolve(1)` but can't `let resolve = Promise.resolve; resolve(1);` but then you can `let resolve = Promise.resolve.bind(Promise); resolve(1);`

Not all object needs to be bound, eg. `let now = Date.now; now();`
